src/web/sound.rs

```diff
@@ -1,15 +1,143 @@
+use wasm_bindgen::{JsValue, JsCast};
+use js_sys::{Error, Array, ArrayBuffer, Map};
+use web_sys::{AudioContext, AudioBuffer, AudioNode, AudioBufferSourceNode, ConstantSourceNode, Response};
+use wasm_bindgen_futures::JsFuture;
 use futures::channel::mpsc::UnboundedReceiver;
+use futures::StreamExt;
+use webutil::event::EventTargetExt;
 
 use crate::sound::SoundCommand;
 
-// TODO
+pub(crate) type InternalSound = AudioBuffer;
 
-pub type InternalSound = ();
+thread_local! {
+    static AUDIO_CONTEXT: AudioContext = AudioContext::new().unwrap();
+}
 
 pub(crate) async fn load(source: &str) -> Result<InternalSound, String> {
-    Ok(())
+    async fn load_(source: &str) -> Result<InternalSound, JsValue> {
+        let response: Response = JsFuture::from(web_sys::window().unwrap().fetch_with_str(source))
+            .await?
+            .dyn_into()
+            .unwrap();
+        let buffer: ArrayBuffer = JsFuture::from(response.array_buffer()?)
+            .await?
+            .dyn_into()
+            .unwrap();
+        let buffer = JsFuture::from(AUDIO_CONTEXT.with(|ctx| ctx.decode_audio_data(&buffer))?)
+            .await?
+            .dyn_into()
+            .unwrap();
+        Ok(buffer)
+    }
+    load_(source).await.map_err(|err: JsValue| err.dyn_into::<Error>().unwrap().to_string().into())
+}
+
+enum ServiceState {
```

Comment:
I think you can adapt this to be at the sound level instead of the service level, and thus match the behaviour of the desktop implementation.